RELEASE/scripts/autoscend.ash

```diff
@@ -5258,42 +5254,67 @@ boolean LX_loggingHatchet()
 
 boolean LX_getStarKey()
 {
-	if (internalQuestStatus("questL10Garbage") < 9 || internalQuestStatus("questL11Shen") < 7)
+	//if we don't want the star key then skip this function
+	if(!get_property("auto_getStarKey").to_boolean())
 	{
 		return false;
 	}
-	if(!get_property("auto_getStarKey").to_boolean())
+
+	//if already have the star key or already used it, then stop running this function.
+	if(!needStarKey())
+	{
+		set_property("auto_getStarKey", false);
+		return false;
+	}
+	
+	//if we did not progress enough in giant garbage quest to reach hole in the sky then skip this function
+	//if shen quest is not far ahead enough also skip. because shen might send us there and it would be a waste of adventures if we already got the star key before he sends us there.
+	if (internalQuestStatus("questL10Garbage") < 9 || internalQuestStatus("questL11Shen") < 7)
 	{
 		return false;
 	}
+	
+	//can we reach the hole in the sky. need rocketship, except in kingdom of exploathing.
 	if(item_amount($item[Steam-Powered Model Rocketship]) == 0 && !in_koe())
 	{
 		return false;
 	}
-	if(!needStarKey())
+	
+	//pull a star chart when all following are true: not hardcore. at the door. no chart. no key. key not used.
```

Comment:
again, add boolean variables whose names are non-obvious conditions

Here, I would say "at the door" is non-obvious, everything else is obvious. Remove comment.